src/main/java/teammates/ui/controller/InstructorCourseInstructorEditSaveAction.java

```diff
@@ -28,7 +29,23 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, account.googleId);
         gateKeeper.verifyAccessible(instructor, logic.getCourse(courseId),
                                     Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR);
-
+        boolean isDisplayedToStudents = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_IS_DISPLAYED_TO_STUDENT) != null;
+        if (!isDisplayedToStudents) {
+            List<InstructorAttributes> instructors = logic.getInstructorsForCourse(courseId);
+            InstructorAttributes instructortoEdit;
+            if (instructorId == null) {
+                instructortoEdit = logic.getInstructorForEmail(courseId, instructorEmail);
+            } else {
+                instructortoEdit = logic.getInstructorForGoogleId(courseId, instructorId);
+            }
+            isError = !logic.isAtLeastOneInstructorVisibleToStudents(instructors, instructortoEdit.email);
```

Comment:
Yes i am thinking i am wasting query by doing like this? Should i go with the solution of adding counter?